src/GeneticSharp.Domain/Populations/Population.cs

```diff
@@ -173,9 +173,10 @@ public virtual void EndCurrentGeneration()
         /// <param name="args">The event arguments.</param>
         protected virtual void OnBestChromosomeChanged(EventArgs args)
         {
-            if (BestChromosomeChanged != null)
+            var handler = BestChromosomeChanged;
```

Comment:
What is the advantage of putting the eventhandler in a variable before comparing it?

Reply:
https://codeblog.jonskeet.uk/2015/01/30/clean-event-handlers-invocation-with-c-6/